lib/generic-pool.js

```diff
@@ -366,23 +366,31 @@ Pool.prototype._ensureMinimum = function _ensureMinimum () {
 
 /**
  * Request a new client. The callback will be called,
- * when a new client will be availabe, passing the client to it.
+ * when a new client will be available, passing the client to it.
  *
  * @param {Function} callback
  *   Callback function to be called after the acquire is successful.
  *   The function will receive the acquired item as the first parameter.
  *
  * @param {Number} priority
- *   Optional.  Integer between 0 and (priorityRange - 1).  Specifies the priority
- *   of the caller if there are no available resources.  Lower numbers mean higher
- *   priority.
+ *   Optional. Integer between 0 and (priorityRange - 1). Specifies the
+ *   priority of the caller if there are no available resources. Lower numbers
+ *   mean higher priority.
+ *
+ * @param {Object} options
+ *   Optional. If `options.block` is `false`, immediately return an error if
+ *   no space is available in the queue.
  *
  * @returns {boolean} `true` if the pool is not fully utilized, `false` otherwise.
  */
-Pool.prototype.acquire = function acquire (callback, priority) {
+Pool.prototype.acquire = function acquire (callback, priority, options) {
   if (this._draining) {
     throw new Error('pool is draining and cannot accept work')
   }
+  if (options !== null && typeof options !== "undefined" && options.block === false && this._count >= this._factory.max) {
```

Comment:
This needs to be `_total`, not `_count`. My fault. I'll add a test for this fix it tomorrow PST.